runelite-client/src/main/java/net/runelite/client/ui/components/ThinProgressBar.java

```diff
@@ -23,54 +23,73 @@
  * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
-package net.runelite.client.plugins.hiscore;
+package net.runelite.client.ui.components;
 
-import java.awt.Graphics;
-import java.awt.Insets;
-import javax.swing.BorderFactory;
-import javax.swing.Icon;
-import javax.swing.JTextField;
-import javax.swing.border.Border;
+import java.awt.BorderLayout;
+import java.awt.Color;
+import java.awt.Dimension;
+import javax.swing.JPanel;
+import lombok.Setter;
 
-public class IconTextField extends JTextField
+/**
+ * A progress bar to be displayed underneath the GE offer item panels
+ */
+public class ThinProgressBar extends JPanel
 {
-	private Border border;
-	private Icon icon;
+	@Setter
+	private int maximumValue;
 
-	@Override
-	public void setBorder(Border border)
+	@Setter
+	private int value;
+
+	private JPanel topBar;
+
+	public ThinProgressBar()
 	{
-		this.border = border;
+		setLayout(new BorderLayout());
+		setBackground(Color.GREEN.darker());
 
-		if (icon == null)
-		{
-			super.setBorder(border);
-		}
-		else
-		{
-			Border margin = BorderFactory.createEmptyBorder(0, icon.getIconWidth() + 4, 0, 0);
-			Border compound = BorderFactory.createCompoundBorder(border, margin);
-			super.setBorder(compound);
-		}
+		topBar = new JPanel();
+		topBar.setPreferredSize(new Dimension(100, 4));
+		topBar.setBackground(Color.GREEN);
+
+		add(topBar, BorderLayout.WEST);
 	}
 
-	@Override
-	public void paintComponent(Graphics graphics)
+	/**
+	 * Updates the UI based on the percentage progress
+	 */
+	public void update()
 	{
-		super.paintComponent(graphics);
+		double percentage = getPercentage();
+		int topWidth = (int) (getSize().width * (percentage / 100));
+
+		topBar.setPreferredSize(new Dimension(topWidth, 4));
+		topBar.repaint();
 
-		Insets iconInsets = border.getBorderInsets(this);
-		icon.paintIcon(this, graphics, iconInsets.left, iconInsets.top);
+		revalidate();
+		repaint();
 	}
 
-	public void setIcon(Icon icon)
+	public double getPercentage()
 	{
-		this.icon = icon;
-		resetBorder();
+		if (value == 0)
+		{
+			return 0;
+		}
+
+		return (value * 100) / maximumValue;
 	}
 
-	private void resetBorder()
+	@Override
+	public void setForeground(Color color)
 	{
-		setBorder(border);
+		if (topBar == null)
```

Comment:
This nullcheck is not neccessary

Reply:
It seems to be, I could't quite understand why, but it seems the setForeground gets called before the topBar is initialized, do you have any idea how this could be?

Reply:
Its probably called in the superclass's constructor somewhere.

Reply:
I could be reading it wrong, but I'm sure it only calls that method after the instance is inicialized.